myapp/templates/single_stock.html

```diff
@@ -61,64 +61,101 @@
-							data: historic_data.map(d => d.close),
-							backgroundColor: [
-								'rgba(105, 0, 132, .2)',
-							],
-							borderColor: [
-								'rgba(200, 99, 132, .7)',
-							],
-							borderWidth: 2
-						}
-					]
-				},
-				options: {
-					responsive: true
-				}
-			});
-		});
-    });
+    $(document).ready(getHistoricData());
+
+    function getHistoricData(time_range='1m') {
+        // reset canvas to avoid conflicting hover events from other graphs
+        var chartContainer = document.getElementById('chartContainer');
+        chartContainer.innerHTML = '';
+        $('#chartContainer').append('<canvas id="stockChart"></canvas>');
+
+        $.get(`/historic/{{ data.symbol }}/${time_range}/`, function ( data, received, response ) {
+            if (response.status == 200) {
+                historic_data = data.data.sort(function(a, b) { return a.date - b.date; })
+
+                // For some strange reason, the 'today' data point returned by API has no label
+                if (typeof historic_data[data.data.length-1].label === 'undefined') {
+                    last_label = new Date(historic_data[data.data.length-1].date).toString();
+                    if (time_range == '3m' || time_range == '6m' || time_range.endsWith('y')){
+                        historic_data[data.data.length-1].label = 
+                            last_label.substr(3, 7) + ', ' 
+                            + historic_data[data.data.length-1].date.substr(2, 2);
+                    }
+                    else{
+                        historic_data[data.data.length-1].label = last_label.substr(3, 7);
+                    }
+                }
+
+                var colors = getGraphColors(data.data[0].close, data.data[data.data.length-1].close);
+                var ctxL = document.getElementById('stockChart').getContext('2d');
+
+                var myLineChart = new Chart(ctxL, {
+                    type: 'line',
+                    data: {
+                        labels: historic_data.map(d => d.label),
+                        datasets: [
+                            {
+                                label: '{{ data.symbol }}',
+                                data: historic_data.map(d => d.close),
+                                backgroundColor: [colors[0]],
+                                borderColor: [colors[1]],
+                                borderWidth: 2
+                            }
+                        ]
+                    },
+                    options: 
+                        {
+                            responsive: true,
+                            spanGaps: true,
+                        }
+                    });
+                }
+        });
+    }
+
+    function getGraphColors(p1, p2) {
+        if (p2 >= p1) {
+            return ['rgba(2, 144, 62, .2)', 'rgba(2, 144, 62, .5)']
+        }
+        else {
+            return ['rgba(220, 24, 24, .2)', 'rgba(220, 24, 24, .5)']
+        }
+    }
```

Comment:
you can use random color to be returned whenever it's called, like this (copied):

> 
> function random_rgba() {
>     var o = Math.round, r = Math.random, s = 255;
>     return 'rgba(' + o(r()*s) + ',' + o(r()*s) + ',' + o(r()*s) + ',' + r().toFixed(1) + ')';
> }

if you would like to add for your and mine (in stocks-comparison) usage.

Reply:
Okay, I will add this to my code